### PR TITLE
[Rector] Apply Rector dead code set list

### PR DESCRIPTION
**Pull request by samsonasik**
Rector dead code set list `SetList::DEAD_CODE` is collection of rector rules to remove dead code at 

https://github.com/rectorphp/rector-src/blob/c3c1411d107534e9d5c103982c9e5442213cc800/config/set/dead-code.php

This PR apply rector dead code set list with use latest rector 0.11.48, with skip some edge cases on purpose or need wait for next release for the false positive check functionality.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: system/Security/Security.php [hunk 1]
```diff
@@ -125,7 +125,9 @@ class Security implements SecurityInterface
      */
     public function __construct(App $config)
     {
-        /** @var SecurityConfig */
+        /**
+         * @var SecurityConfig $security
+         */
```
Comment (MGatner) [inline system/Security/Security.php:130]: How come this was forced to multi-line when the same thing in **Events.php** reverted to a single line?

File: system/Security/Security.php [hunk 2]
```diff
@@ -134,7 +136,9 @@ public function __construct(App $config)
         $this->regenerate = $security->regenerate ?? $config->CSRFRegenerate ?? $this->regenerate;
         $rawCookieName    = $security->cookieName ?? $config->CSRFCookieName ?? $this->cookieName;
 
-        /** @var CookieConfig */
+        /**
+         * @var CookieConfig $cookie
+         */
         $cookie = config('Cookie');
```
Comment (MGatner) [inline system/Security/Security.php:138]: Ditto.

File: system/Session/Session.php [hunk 3]
```diff
@@ -182,7 +182,9 @@ public function __construct(SessionHandlerInterface $driver, App $config)
         $this->cookieSecure   = $config->cookieSecure ?? $this->cookieSecure;
         $this->cookieSameSite = $config->cookieSameSite ?? $this->cookieSameSite;
 
-        /** @var CookieConfig */
+        /**
+         * @var CookieConfig $cookie
+         */
```
Comment (MGatner) [inline system/Session/Session.php:187]: Ditto.

File: tests/system/Session/SessionTest.php [hunk 4]
```diff
@@ -180,9 +180,9 @@ public function testGetReturnsItemValueisZero()
         $session = $this->getInstance();
         $session->start();
 
-        $session->set('foo', (int) 0);
```
Comment (MGatner) [inline tests/system/Session/SessionTest.php:183]: That's just funny. 🤗

**Review by MGatner — APPROVED** [review]
Looks good! Mostly useless casts and incomplete or unnecessary docblocks. I'm not sure why some inline docblocks are multi-line and others are single but I would like to see that standardized.

**Review by MGatner — CHANGES_REQUESTED** [review]
I swear I hit "Request Changes". GitHub Mobile must be trolling me.

File: system/Security/Security.php (hunk 1 above)
Comment (samsonasik) [reply system/Security/Security.php]: this change is my manual change as it remove `@var` due to no variable definition after `@var` type. Updated to single line.

File: system/Security/Security.php (hunk 2 above)
Comment (samsonasik) [reply system/Security/Security.php]: fixed.

File: system/Session/Session.php (hunk 3 above)
Comment (samsonasik) [reply system/Session/Session.php]: fixed.

**Review by MGatner — APPROVED** [review]

**Comment by samsonasik** [comment]
@MGatner thank you for the review.

**Pull request merged**